src/script_opt/CPP/Types.cc

```diff
@@ -189,9 +189,37 @@ shared_ptr<CPP_InitInfo> CPPCompile::RegisterType(const TypePtr& tp) {
 
     processed_types[t] = nullptr;
 
+    // Only applicable for records. If non-zero, reflects where - for
+    // standalone compilation - redef+= record fields start.
+    int addl_fields = 0;
+    bool type_init_needed = standalone && obj_matches_opt_files(tp);
+
+    if ( standalone && ! type_init_needed ) {
+        if ( tp->Tag() == TYPE_RECORD ) {
+            auto tr = tp->AsRecordType();
+            for ( auto i = tr->NumOrigFields(); i < tr->NumFields(); ++i ) {
+                auto fd = tr->FieldDecl(i);
+                if ( filename_matches_opt_files(fd->loc.filename) ) {
+                    if ( addl_fields == 0 )
+                        addl_fields = i;
+                }
+                else if ( addl_fields > 0 ) {
+                    reporter->Error(
```

Comment:
Seems `FatalError()` works for Error + exit(1).

I'm not 100% sure what's going on here. If a user forgets to add a file that redefs to `only_files`, this triggers an error? It would probaby be helpful to output the filename from `fd->loc.filename` in the message?

Reply:
Yeah you're right it's fairly baffling. I added a comment above to explain the considerations, and expanded the error message to include the name of the record with the issue. (I'll add the filename too if it subsequently proves helpful, not sure whether it's needed given the record name.)